### ISSUE TITLE
TodayButton is not displayed as a caption

### ISSUE THREAD

**Issue by saenglert** [gpbl/react-day-picker#527]
Hello again,
I've put some more thought into displaying the `.DayPicker-TodayButton` as caption instead of footer (see #523) and been digging through your code a little bit more (no dirty laundry found yet :) ).

From a UI perspective it would clearly be beneficial to have the button at the top of the calendar where also the control for the months are which take you furthest in terms of time forwards or backwards.

With this in mind I looked at why the `todayButton` was behaving strangely as a caption and found that there already had been a caption assigned to the table via CSS namely the `.DayPicker-Caption` which contains the name of the month currently viewed by the calendar.

Apparently it is not possible for a table to have multiple stacked captions. However removing the style property pushes the text into the grid defined by `.DayPicker-Month` so I looked for a place to store it outside the table.

One thing that struck me as odd when I first fired up DayPicker was the way you build the `.DayPicker-NavBar`. Visually it forms one piece with `.DayPicker-Caption` but it's actually moved there via absolute positioning.

I guess there's a practical reason for this?

Anyway, so I stuck the `.DayPicker-Caption` into `.DayPicker-NavBar` and was able to pretty easily piece together the visuals via disabling the absolute positions of the `.DayPicker-Navbar` and it's buttons replacing it with a flex construct.

With the `.DayPicker-Caption` out of the way the `todayButton` now can easily switch between places at the top and bottom. In fact it doesn't even need the fix I added in #523 anymore because this can be controlled via the `caption-side` CSS property as you already intended to do. 

So in the future switching the position from top to bottom could be as simple as changing the state of the component to `top` or `bottom`.

I've not yet implemented the changes in hard code cause to be quite honest I got no idea how to build and test changes in a npm project/package (University student here on his first internship *\*insert jon snow knows nothing meme*\*).

So anyways I hope to have given you some ideas on what could potentially use some work.

**Comment by gpbl** [comment]
> From a UI perspective it would clearly be beneficial to have the button at the top of the calendar where also the control for the months are which take you furthest in terms of time forwards or backwards.

Thanks for the suggestion!
Do you have a screenshot to show what do you mean?

Anyway, we aren’t going to fit everyone’s needs here. If you need a different layout, you can always play woth the `captionElement` prop. Or wrap the component in another component.

> so I looked for a place to store it outside the table

Why not put the day-picker in another component? or are you referring to the input’s overlay?

> One thing that struck me as odd when I first fired up DayPicker was the way you build the `.DayPicker-NavBar`. Visually it forms one piece with `.DayPicker-Caption` but it's actually moved there via absolute positioning.

Still is semantically its caption, even if absolutely positioned. :) 
The reason I didn’t use flex-box there is mainly for browsers compatibility!

thanks for your feedback !

**Comment by saenglert** [comment]
I made a quick sketch, hope it gets the point across:

![todaybuttonplacement](https://user-images.githubusercontent.com/6491588/31931753-7f9cdea0-b8a4-11e7-9cac-2d8827f600ca.png)

> Anyway, we aren’t going to fit everyone’s needs here. If you need a different layout, you can always play with the captionElement prop. Or wrap the component in another component.

Yeah probably not, but it's the only thing that's not been working out of the box and our UX guy demands his today button up there :) . 

I'll try out adding a custom `captionElement`, thanks for the tip!

> Why not put the day-picker in another component? or are you referring to the input’s overlay?

Guess I got you confused there, sorry: When removing the `display: table-caption` property from the `captionElement` it's default behaviour is to turn into a single table cell because it is inside the `Month` component. By moving it into the navigation bar I was able to stretch it out across the whole top of the table again. `display: table` and other display options don't seem to get along very well.

![screenshot-2017-10-24 mc schedule](https://user-images.githubusercontent.com/6491588/31932376-756872b2-b8a6-11e7-9cbe-9842858610db.png)
